lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2135,26 +2135,57 @@ namespace Js
     {
         charcount_t count = pThis->GetLength();
 
-        BufferStringBuilder builder(count, pThis->type->GetScriptContext());
-        const wchar_t *inStr = pThis->GetString();
-        wchar_t *outStr = builder.DangerousGetWritableBuffer();
+        const wchar_t* inStr = pThis->GetString();
+        const wchar_t* i = inStr; 
+        const wchar_t* iLim = i + count;
+        
+        // first, try to see if the string needs recasing at all
+        while(i < iLim)
```

Comment:
`while` and `if` statements should have a space between the keyword and the `(`, to match the style elsewhere in this file.